### DISCUSSION TITLE
Issues with moving from current Data Model XML format to the new Data Model XML

### DISCUSSION THREAD

**Discussion by brdandu** [project-chip/zap#1372] [General topics]
This page discusses the issues related to porting from existing matter repo xml to the new xml

**Comment by brdandu** [comment]
Issues with Attribute porting:
- The integer types for attributes have changed format for eg: int16u is now uint16 -> Can this be fixed in the Spec to XML conversion?
- Data types such as char_string,long_char_string(octet) have changed to just string for attribute types -> Does the implementation determine how to figure out string lengths when passed in command arguments when just denoted as strings?
- Attribute type names which had `_` have now switched to `-` eg: vendor_id to vendor-id -> Can this be fixed in the Spec to XML conversion?
- Attribute type="array" has become type="list" -> We should probably fix this in ZAP such that it can handle list types as well.
- The way entryType is mentioned in new xml has changed. In the new xml entry type is a sub tag instead of just an attribute. -> Can this be fixed in the Spec to XML conversion?

**Comment by cecille** [comment]
1) spec uses uint16 style. Is int16u a zap requirement?
2) is long_char_string(octet) the zap code for an octet string?
3) yeah, can probably fix in alchemy once we swap
4) ok
5) entry tag is for the types for lists, so we can't have a second "type" attribute on the `<attribute>` tag. Do you want entryType attribute, but only populated on tags with the type of list?

**Comment by paulr34** [comment]
@cecille there is no Endpoint Composition in the new XML https://github.com/project-chip/connectedhomeip/issues/32941
